### PR TITLE
virttest.virt_vm: optimize wait_for_serial_login function

### PR DESCRIPTION
**Pull request by xutian**
Signed-off-by: Xu Tian <xutian@redhat.com>

### CODE REVIEW

**Comment by xutian** [comment]
@PandaWei  would you like to help review this PR. 

Thanks,
Xu

**Comment by PandaWei** [comment]
You want to distinguish LoginError and xxxError introduced by restarting network , and interrupt and raise error if the latter occurs?

**Comment by xutian** [comment]
@PandaWei , yes, I think current logical looks more clearly and more easy to understand and after this update there will no such many useless debug information recording in log-file during guest bootup.

File: virttest/virt_vm.py [hunk 1]
```diff
@@ -1206,33 +1208,28 @@ def wait_for_serial_login(self, timeout=LOGIN_WAIT_TIMEOUT,
...
+                    utils_net.restart_guest_network(session, os_type=os_type)
                 return session
             except remote.LoginError, e:
+                exception = e
                 self.verify_alive()
-                e = str(e)
-                if e not in error_messages:
-                    logging.debug(e)
-                    error_messages.append(e)
-            time.sleep(2)
```
Comment (PandaWei) [inline virttest/virt_vm.py]: Here I want keep to output the error messages, and sleep 2.
error message is useful and 0.5 is short.

wait_for_login has the same code with wait_for_serial_login.
It's necessary to keep consistence.

File: virttest/virt_vm.py [hunk 2]
```diff
@@ -1206,33 +1208,28 @@ def wait_for_serial_login(self, timeout=LOGIN_WAIT_TIMEOUT,
...
-                    os_type = self.params.get("os_type")
-                    try:
-                        logging.debug("Attempting to restart guest network")
-                        utils_net.restart_guest_network(session,
-                                                        os_type=os_type)
-                    except Exception:
-                        pass
+                    logging.debug("Attempting to restart guest network")
+                    os_type = self.params.get('os_type')
+                    utils_net.restart_guest_network(session, os_type=os_type)
```
Comment (PandaWei) [inline virttest/virt_vm.py]: Here catching shellError and ExceptError  looks more clear

File: virttest/virt_vm.py [hunk 3]
```diff
@@ -1206,33 +1208,28 @@ def wait_for_serial_login(self, timeout=LOGIN_WAIT_TIMEOUT,
         :params virtio: is a virtio console.
         :return: A ShellSession object.
         """
-        error_messages = []
         logging.debug("Attempting to log into '%s' via serial console "
                       "(timeout %ds)", self.name, timeout)
-        end_time = time.time() + timeout
+        end_time, exception = time.time() + timeout, None
         while time.time() < end_time:
             try:
```
Comment (PandaWei) [inline virttest/virt_vm.py]: Does need to reset exception to None?

Comment (xutian) [inline virttest/virt_vm.py]: var exception just used to keep the latest Exception during login VM.

File: virttest/virt_vm.py (hunk 2 above)
Comment (xutian) [inline virttest/virt_vm.py]: do you mean use nested try...except ?

File: virttest/virt_vm.py (hunk 1 above)
Comment (xutian) [inline virttest/virt_vm.py]: Agree with keep sleep time to 2s. keep error message looks useless I think information we only need to care is exception raised in latest.
Will open a new PR to update wait_for_login function to keep consistence.

Thanks,
Xu

File: virttest/virt_vm.py (hunk 3 above)
Comment (PandaWei) [inline virttest/virt_vm.py]: Yes, That's why I ask to reset exception to None.
If the "N-1" time failed and set exception, but "N" time success,
the exception is not correct.
 I'm right?

File: virttest/virt_vm.py (hunk 2 above)
Comment (PandaWei) [inline virttest/virt_vm.py]: Yes

File: virttest/virt_vm.py (hunk 3 above)
Comment (xutian) [inline virttest/virt_vm.py]: If N time sucess, session will return and exception the last line will not be execute.

**Comment by PandaWei** [comment]
I'm OK for this PR.

**Comment by xutian** [comment]
Merged, thanks @PandaWei

**Pull request merged**